app/lib/exchange-rate/providers/coinbase.ts

```diff
@@ -0,0 +1,36 @@
+import type {
+  ExchangeRateProvider,
+  GetRatesParams,
+  Rates,
+} from '~/lib/exchange-rate/providers/types';
+
+export class Coinbase implements ExchangeRateProvider {
+  supportedTickers: string[] = ['BTC-USD'];
+
+  async getRates({ tickers, signal }: GetRatesParams): Promise<Rates> {
+    if (!tickers.length) {
+      throw new Error('No tickers provided');
+    }
+
+    tickers.forEach((ticker) => {
+      if (!this.supportedTickers.includes(ticker)) {
+        throw new Error(`Unsupported ticker: ${ticker}`);
+      }
+    });
+
+    const response = await fetch(
+      'https://api.coinbase.com/v2/prices/BTC-USD/spot',
```

Comment:
I just realized that coinbase also has this endpoint `https://docs.cdp.coinbase.com/coinbase-app/docs/api-exchange-rates` that returns more tickers. We should change to use that. We can ofc ignore all the shitcoin rates they return there and just pick ones we think we might need